api/v1alpha1/flowcollector_types.go

```diff
@@ -201,6 +201,37 @@ type FlowCollectorKafka struct {
 	TLS ClientTLS `json:"tls"`
 }
 
+const (
+	PrometheusTLSDiasbled = "DISABLED"
+	PrometheusTLSManual   = "MANUAL"
+	PrometheusTLSAuto     = "AUTO"
+)
+
+type PrometheusTLSConfigType string
+
+// PrometheusConfig define the prometheus endpoint configuration
+type PrometheusConfig struct {
+
+	//+kubebuilder:validation:Minimum=1
+	//+kubebuilder:validation:Maximum=65535
+	//+kubebuilder:default:=9090
+	// the prometheus HTTP port
+	Port int32 `json:"port,omitempty"`
+
+	// Select the type of TLS configuration
+	// "DISABLED" (default) to not configure TLS for the endpoint, "MANUAL" to manually provide cert file and a key file,
+	// and "AUTO" to use Openshift auto generated certificate using annotations
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum:="DISABLED";"MANUAL";"AUTO"
+	// +kubebuilder:validation:Required
+	//+kubebuilder:default:="DISABLED"
+	TLSType PrometheusTLSConfigType `json:"tlsType,omitempty"`
+
+	// TLS configuration.
+	// +optional
+	ManualTLS ClientTLS `json:"manualTls"`
```

Comment:
Maybe we should not reuse `ClientTLS` here, and create a new struct, I believe it is not 100% relevant when used to configure tls on server side (or at least it should be renamed).
As @jpinsonneau pointed, `ClientTLS.enable` setting can be confusing as we have `TLSType` here. Also I believe InsecureSkipVerify is not relevant on server side?

Reply:
Yes, after Julien comment, I changed the code to use CertificateReference instead of ClientTLS